app/iiif_print/metadata_decorator.rb

```diff
@@ -0,0 +1,119 @@
+module IiifPrint
+  module MetadataDecorator
+    private
+
+      def faceted_values_for(field_name:)
+        values_for(field_name: field_name).map do |value|
+          # In BL, creator facet is `creator_search_sim`
+          search_field = field_name == :creator ? field_name.to_s + "_search_sim" : field_name.to_s + "_sim"
+          path = Rails.application.routes.url_helpers.search_catalog_path(
+            "f[#{search_field}][]": value, locale: I18n.locale
+          )
+          path += '&include_child_works=true' if work["is_child_bsi"] == true
+          "<a href='#{File.join(@base_url, path)}'>#{value}</a>"
+        end
+      end
+
+      # BL has three special name fields that are split into
+      # family name, given name, and also organization name
+      # which need to be handled differently
+      SPECIAL_NAME_FIELDS = [:creator, :contributor, :editor].freeze
+
+      def values_for(field_name:)
+        field_name = field_name.try(:name) || field_name
+        values = Array(work["#{field_name}_tesim"] || work["#{field_name}_dtsi"]&.to_date.try(:to_formatted_s, :standard))
+        return if values.empty?
+
+        if SPECIAL_NAME_FIELDS.include?(field_name)
+          handle_names(field_name, values)
+        elsif field_name == :funder
+          handle_funder(values)
+        elsif field_name == :alternate_identifier
+          handle_alternative_identifier(values)
+        elsif field_name == :related_identifier
+          handle_related_identifier(values)
+        elsif field_name == :resource_type
+          handle_resource_type(values)
+        else
+          values
+        end
+      end
+
+      def handle_names(field_name, values)
+        return unless Ubiquity::JsonValidator.valid_json?(values.first)
+
```

Comment:
Consider parsing with Ubiquity::ParseJson class instead of duplicating that logic.

Reply:
Cool find!  Sadly, this only worked for creator and not contributor nor editor.